libs/core/src/main/java/org/elasticsearch/common/unit/TimeValue.java

```diff
@@ -182,63 +182,9 @@ public double getDaysFrac() {
         return daysFrac();
     }
 
-    /**
-     * Returns a {@link String} representation of the current {@link TimeValue}.
-     *
-     * Note that this method might produce fractional time values (ex 1.6m) which cannot be
-     * parsed by method like {@link TimeValue#parse(String, String, String)}.
-     */
     @Override
     public String toString() {
```

Comment:
I understand the reason of this change but we can't just replace the toString() method with getStringRep() as it would break the xcontent rendering of "human readable" timevalues.

This is what is rendered today for a timevalue when the `?human` parameter is set for cluster stats:
```
"max_uptime" : "8.3m",
"max_uptime_in_millis" : 501911,
```

with such change it would become:

```
"max_uptime" : "501911ms",
"max_uptime_in_millis" : 501911,
```

Maybe we could have two methods, something like `toHumanReadableString()` and the regular `toString()` (which implementation is copied from getStringRep()) and adapt the xcontent date formatters accordingly? (See XContentElasticsearchExtension)

Reply:
Ah, I forgot about human-readable. Nevermind then, I think its okay as it is then. Thanks for the explanation, will close this then.